logcat/build.gradle.kts

```diff
@@ -1,22 +1,87 @@
 plugins {
+  kotlin("multiplatform")
   id("com.android.library")
-  kotlin("android")
-  id("com.vanniktech.maven.publish")
+}
+
+repositories {
+  mavenCentral()
+  gradlePluginPortal()
+  google()
+}
+
+kotlin {
+  jvm {
+    compilations.all {
+      kotlinOptions.jvmTarget = "1.8"
+    }
+  }
+  js(IR) {
+    browser {
+      commonWebpackConfig {
+        cssSupport.enabled = true
+      }
+    }
+    binaries.executable()
+  }
+  macosX64("native") {
+    binaries {
+      sharedLib {
+        baseName = "logcat"
+      }
+      framework {
+        baseName = "Logcat"
+      }
+    }
+  }
+  val hostOs = System.getProperty("os.name")
+  val isMingwX64 = hostOs.startsWith("Windows")
+  val nativeTarget = when {
+    hostOs == "Mac OS X" -> macosX64("native")
+    hostOs == "Linux" -> linuxX64("native")
+    isMingwX64 -> mingwX64("native")
```

Comment:
Typically we include all native targets. Also this is missing many important targets like ARM Macs and iOS.

Reply:
I have added more targets that will come in the next push.  I created the `nativeMain` and `nativeTest` source sets and made the other targets dependOn them.  I only have a Mac, so I could not build some of the targets